test/org/loklak/api/search/GithubProfileScraperTest.java

```diff
@@ -1,45 +1,81 @@
+        githubScraper.termsList = new ArrayList<String>();
+        githubScraper.termsList.add("all");
+
+        try {
+            ClientConnection connection = new ClientConnection(url);
+            br = githubScraper.getHtml(connection);
+
+            String profile = "fossasia";
+            String shortDescription = "Open Technologies in Asia";
+            String userName = "fossasia";
+            String userId = "6295529";
+            String location = "Singapore";
+            String specialLink = "http://fossasia.org";
+
+    	    Post fetchedProfile = githubScraper.scrapeGithub(profile, br);
+
+            assertEquals(fetchedProfile.getString("short_description"), shortDescription);
+            assertEquals(fetchedProfile.getString("user_name"), userName);
+            assertEquals(fetchedProfile.getString("user_id"), userId);
+            assertEquals(fetchedProfile.getString("location"), location);
+            assertEquals(fetchedProfile.getString("special_link"), specialLink);
+        } catch (IOException e) {
+            DAO.log("GithubProfileScraperTest.githubProfileScraperUserTest() failed to connect to network. url:" + url);
+        }
+    }
 
 	@Test
-	public void githubProfileScraperUserTest() {
-		String profile = "djmgit";
-		String userName = "djmgit";
-		String fullName = "Deepjyoti Mondal";
-		String specialLink = "http://djmgit.github.io";
-		String userId = "16368427";
-
-		SusiThought response = GithubProfileScraper.scrapeGithub(profile);
-		JSONObject fetchedProfile = (JSONObject)response.getData().get(0);
-
-		assertEquals(fetchedProfile.getString("user_name"), userName);
-		assertEquals(fetchedProfile.getString("full_name"), fullName);
-		assertEquals(fetchedProfile.getString("special_link"), specialLink);
-		assertEquals(fetchedProfile.getString("user_id"), userId);
-	}
+    public void githubProfileScraperUserTest() {
+
+        GithubProfileScraper githubScraper = new GithubProfileScraper();
+        String url = "https://github.com/djmgit/";
+        BufferedReader br = null;
+        githubScraper.termsList = new ArrayList<String>();
+        githubScraper.termsList.add("all");
+        
+        try {
+            ClientConnection connection = new ClientConnection(url);
+            br = githubScraper.getHtml(connection);
+
+            String profile = "djmgit";
+            String userName = "djmgit";
+            String fullName = "Deepjyoti Mondal";
+            String specialLink = "http://djmgit.github.io";
+            String userId = "16368427";
+
+            Post fetchedProfile = githubScraper.scrapeGithub(profile, br);
+
+            assertEquals(fetchedProfile.getString("user_name"), userName);
+            assertEquals(fetchedProfile.getString("full_name"), fullName);
+            assertEquals(fetchedProfile.getString("special_link"), specialLink);
+            assertEquals(fetchedProfile.getString("user_id"), userId);
+        } catch (IOException e) {
```

Comment:
`IOException` is needed to be handled as ClientConnection throws this error. If there is IOException, it is clear that there is network error